storage/client_test.go

```diff
@@ -124,18 +127,126 @@ func TestGetSetTestIamPolicyEmulated(t *testing.T) {
 	})
 }
 
+func TestListObjectsEmulated(t *testing.T) {
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
+		// Populate test data.
+		_, err := client.CreateBucket(context.Background(), project, &BucketAttrs{
+			Name: bucket,
+		})
+		if err != nil {
+			t.Fatalf("client.CreateBucket: %v", err)
+		}
+		prefix := time.Now().Nanosecond()
+		want := []*ObjectAttrs{
+			{
+				Bucket: bucket,
+				Name:   fmt.Sprintf("%d-object-%d", prefix, time.Now().Nanosecond()),
+			},
+			{
+				Bucket: bucket,
+				Name:   fmt.Sprintf("%d-object-%d", prefix, time.Now().Nanosecond()),
+			},
+			{
+				Bucket: bucket,
+				Name:   fmt.Sprintf("object-%d", time.Now().Nanosecond()),
+			},
+		}
+		for _, obj := range want {
+			w := veneerClient.Bucket(bucket).Object(obj.Name).NewWriter(context.Background())
+			if _, err := w.Write(randomBytesToWrite); err != nil {
+				t.Fatalf("failed to populate test data: %v", err)
+			}
+			if err := w.Close(); err != nil {
+				t.Fatalf("closing object: %v", err)
+			}
+		}
+
+		// Simple list, no query.
+		it := client.ListObjects(context.Background(), bucket, nil)
+		var o *ObjectAttrs
+		for i := 0; err == nil && i <= len(want); i++ {
+			o, err = it.Next()
+			if err != nil {
```

Comment:
Don't totally understand why we are continuing to the next iteration on a non-nil error. In general I would expect this test to check that the number of objects iterated lines up correctly with what's expected based on input.

Reply:
Yeah I will just `break` here. 

> I would expect this test to check that the number of objects iterated lines up correctly

For sure, I'll add this.

Reply:
Adding this actually caught a bug in the emulator list buckets impl for http prefix filtering! lol need https://github.com/googleapis/storage-testbench/pull/317 to land for CI to pass

Reply:
I'm going to move this change to the list buckets test to a separate PR to not block this one